frontend/src/app/modules/users/users-selectors.js

```diff
@@ -1,14 +1,65 @@
 import { createSelector } from 'reselect'
 
+import * as Filters from '../../const/leaderboards-filters'
+import { didUserPlayMatch } from '../matches/matches-selectors'
+import {
+  computeLongestWinStreak,
+  computeWinRatio,
+} from '../matches/matches-computations'
+
 export const getUsers = state => state.users
+const getMatches = state => state.matches
+const getFilters = state => state.filters
 
 export const getUser = createSelector(
   getUsers,
   (stats, userId) => userId,
   (users, userId) => users.find(user => user.id === userId),
 )
 
-export const getTopUsers = createSelector(
+export const getTopRatedUsers = createSelector(
   getUsers,
   users => [...users].sort((user1, user2) => user2.rating - user1.rating),
 )
+
+export const getTopUsers = createSelector(
+  getUsers,
+  getMatches,
+  getFilters,
+  (users, matches, filters) => sortTopUsers([...users], [...matches], filters),
```

Comment:
Why do you spread `users` and `matches`?

Reply:
I took that from the former code without thinking about it. Will refactor it.

Reply:
Fixed [here](https://github.com/salsita/foosball-rating/pull/92/commits/d7c17da31ec421acbb90f4a19ce22c34412cdf35#diff-6bb144976a04c9e3de620e0c0828e32e) and [here](https://github.com/salsita/foosball-rating/pull/92/commits/d7c17da31ec421acbb90f4a19ce22c34412cdf35#diff-08117b849ecc00af7c3c0d609dd86d31)